index.js

```diff
@@ -227,6 +228,242 @@ async function bundle(items, options) {
+
+	const stylesheet = resolve(options.style || './templates/default.css');
+	const style = fs.readFileSync(stylesheet, 'utf8') + (options.css || '');
+
+	const html = nunjucks.renderString(
+		fs.readFileSync(
+			resolve(options.template || './templates/default.html'),
+			'utf8'
+		),
+		{
+			items,
+			style,
+			stylesheet // deprecated
+		}
+	);
+
+	const doc = new JSDOM(html).window.document;
+	const headerTemplate = doc.querySelector('.header-template');
+	const footerTemplate = doc.querySelector('.footer-template');
+	const header = new JSDOM(
+		headerTemplate ? headerTemplate.innerHTML : '<span></span>'
+	).window.document;
+	const footer = new JSDOM(
+		footerTemplate ? footerTemplate.innerHTML : '<span></span>'
+	).window.document;
+
+	const css_ast = css.parse(style);
+
+	const header_style = get_style_attribute_value(css_ast, '.header-template');
+	const header_div = header.querySelector('body :first-child');
+
+	if (header_div && header_style) {
+		header_div.setAttribute(
+			'style',
+			`
+				${header_style};
+				${header_div.getAttribute('style') || ''}
+			`
+		);
+	}
+
+	const footer_style = get_style_attribute_value(css_ast, '.footer-template');
+	const footer_div = footer.querySelector('body :first-child');
+
+	if (footer_div && footer_style) {
+		footer_div.setAttribute(
+			'style',
+			`
+				${footer_style};
+				${footer_div.getAttribute('style') || ''}
+			`
+		);
+	}
+
+	fs.writeFileSync(temp_file, html);
+
+	spinner.succeed(`Temporary HTML file: file://${temp_file}`);
+
+	spinner.start('Saving EPUB');
+
+	const browser = await pup.launch({
+		headless: true,
+		/*
+			Allow running with no sandbox
+			See: https://github.com/danburzo/percollate/issues/26
+		 */
+		args: options.sandbox
+			? undefined
+			: ['--no-sandbox', '--disable-setuid-sandbox'],
+		defaultViewport: {
+			// Emulate retina display (@2x)...
+			deviceScaleFactor: 2,
+			// ...but then we need to provide the other
+			// viewport parameters as well
+			width: 1920,
+			height: 1080
+		}
+	});
+	const page = await browser.newPage();
+	await page.goto(`file://${temp_file}`, { waitUntil: 'load' });
+
+	/*
+		When no output path is present,
+		produce the file name from the web page title
+		(if a single page was sent as argument),
+		or a timestamped file (for the moment)
+		in case we're bundling many web pages.
+	 */
+	const output_path =
+		options.output ||
+		(items.length === 1
+			? `${slugify(items[0].title || 'Untitled page')}.epub`
+			: `percollate-${Date.now()}.epub`);
+
+	let bodyHTML = await page.evaluate(() => document.body.innerHTML);
+
+	let option = {
+		title: items[0].title,
+		content: [
+			{
+				data: bodyHTML
+			}
+		]
+	};
+
+	new Epub(option, output_path);
+
+	await browser.close();
+
+	spinner.succeed(`Saved EPUB: ${output_path}`);
+}
+
+/*
+	Bundle the HTML files into a HTML
+	--------------------------------
+ */
+async function bundleHtml(items, options) {
```

Comment:
Same as for EPUB generation, when we generate the HTML, we don't need to create a temporary HTML file and launch Puppeteer with it. We should be able to just bundle the result of `nunjucks.renderString`.